SUMMARY.md

```diff
@@ -10,31 +10,33 @@
   - [Finalisation in Destructors](./idioms/dtor-finally.md)
   - [`mem::{take(_), replace(_)}`](./idioms/mem-replace.md)
   - [On-Stack Dynamic Dispatch](./idioms/on-stack-dyn-dispatch.md)
-  - [Foreign function interface usage](./idioms/ffi-intro.md)
-    - [Idiomatic Errors](./idioms/ffi-errors.md)
-    - [Accepting Strings](./idioms/ffi-accepting-strings.md)
-    - [Passing Strings](./idioms/ffi-passing-strings.md)
+  - [Foreign function interface (FFI)](./idioms/ffi/intro.md)
+    - [Idiomatic Errors](./idioms/ffi/errors.md)
+    - [Accepting Strings](./idioms/ffi/accepting-strings.md)
+    - [Passing Strings](./idioms/ffi/passing-strings.md)
   - [Iterating over an `Option`](./idioms/option-iter.md)
   - [Pass Variables to Closure](./idioms/pass-var-to-closure.md)
   - [Privacy For Extensibility](./idioms/priv-extend.md)
   - [Easy doc initialization](./idioms/rustdoc-init.md)
   - [Temporary mutability](./idioms/temporary-mutability.md)
 
 - [Design Patterns](./patterns/index.md)
-  - [Builder](./patterns/builder.md)
-  - [Compose Structs](./patterns/compose-structs.md)
-  - [Entry API](./patterns/entry.md)
-  - [Foreign function interface usage](./patterns/ffi-intro.md)
-    - [Object-Based APIs](./patterns/ffi-export.md)
-    - [Type Consolidation into Wrappers](./patterns/ffi-wrappers.md)
-  - [Fold](./patterns/fold.md)
-  - [Interpreter](./patterns/interpreter.md)
-  - [Newtype](./patterns/newtype.md)
-  - [RAII Guards](./patterns/RAII.md)
-  - [Prefer Small Crates](./patterns/small-crates.md)
-  - [Strategy](./patterns/strategy.md)
-  - [Contain unsafety in small modules](./patterns/unsafe-mods.md)
-  - [Visitor](./patterns/visitor.md)
+  - [Behavioural](./patterns/behavioural/intro.md)
+    - [Interpreter](./patterns/behavioural/interpreter.md)
+    - [Newtype](./patterns/behavioural/newtype.md)
+    - [RAII Guards](./patterns/behavioural/RAII.md)
+    - [Strategy](./patterns/behavioural/strategy.md)
+    - [Visitor](./patterns/behavioural/visitor.md)
+  - [Creational](./patterns/creational/intro.md)
+    - [Builder](./patterns/creational/builder.md)
+    - [Fold](./patterns/creational/fold.md)
+  - [Structural](./patterns/structural/intro.md)
+    - [Compose Structs](./patterns/structural/compose-structs.md)
+    - [Prefer Small Crates](./patterns/structural/small-crates.md)
+    - [Contain unsafety in small modules](./patterns/structural/unsafe-mods.md)
```

Comment:
@MarcoIeni Could you check if I sorted them right in here?

Reply:
you forgot Entry API

Reply:
it's an empty template, we forgot this in there:
https://rust-unofficial.github.io/patterns/patterns/entry.html

Reply:
oh, sorry. The rest seems correct.